ivy/functional/ivy/experimental/losses.py

```diff
@@ -154,6 +154,59 @@ def l1_loss(
         return ivy.inplace_update(out, loss) if out is not None else loss
 
 
+@handle_exceptions
+@handle_nestable
+@handle_array_like_without_promotion
+@inputs_to_ivy_arrays
+@handle_array_function
+def l2_loss(
+    input: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    reduction: Optional[str] = "sum",
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    Computes half the L2 norm without the sqrt:
+
+    Parameters
+    ----------
+    input : Union[ivy.Array, ivy.NativeArray]
+        Input array containing input values.
+    reduction : str, optional
+        Reduction method for the output loss. Options:
+        "none" (no reduction), "mean" (mean of losses),
+        "sum" (sum of losses). Default: "mean".
+    out : Optional[ivy.Array], optional
```

Comment:
[ref](https://unify.ai/docs/ivy/overview/deep_dive/docstrings.html#:~:text=This%20is%20a%20good%20starting%20point.%20But%20we%20need%20to%20make%20some%20changes.%20Firstly%2C%20given%20that%20we%20are%20using%20type%20hints%2C%20repeating%20all%20of%20the%20types%20also%20in%20the%20docs%20would%20be%20a%20needless%20duplication.)
```suggestion
    out
```